### PR TITLE
Enhancement/request logs

### LINKED ISSUE #67: Extend debug info with HTTP requests and response
In order to simplify the debugging of the new issues consider to improve debug logs as so they explicitly note incoming HTTP requests and outgoing HTTP responses. There could be a filter which allows to specify which requests/responses should not be logged (e.g. `/`, `/info` and so on)

### PR DESCRIPTION
**Pull request by LyzeOfKiel**
Changes:
* Request data is logged to console, as well as to a file. You can specify ignored routes (for console logging), space separated, via `RELAYER_LOG_IGNORE_ROUTES` env, e.g. `/fee /params/hash/tx`. Also, I removed some common headers from requests to not flood the logs. Example request log:
```
info: HTTP GET /params/hash/tree {"meta":{"req":{"headers":{"host":"127.0.0.1:8000","user-agent":"PostmanRuntime/7.30.0"},"httpVersion":"1.1","method":"GET","originalUrl":"/params/hash/tree","query":{},"url":"/params/hash/tree"},"res":{"statusCode":200},"responseTime":1}}
```
Closes #67 

This PR should be considered as a draft, so any suggestions about logs format are welcome.

### CODE REVIEW

**Comment by LyzeOfKiel** [comment]
Updates:
* Set level to `debug`
* Removed duplicated properties from request log, such as `url`, `originalUrl`, `query`, `method`

**Review by r0wdy1 — APPROVED** [review]

File: zp-relayer/index.ts [hunk 1]
```diff
@@ -2,13 +2,14 @@ import './env'
...
 import { logger } from './services/appLogger'
-import { createLoggerMiddleware } from './services/loggerMiddleware'
+import { createConsoleLoggerMiddleware, createPersistentLoggerMiddleware } from './services/loggerMiddleware'
 import config from './config'
 import { init } from './init'
 
 const app = express()
 
-app.use(createLoggerMiddleware('zp.log'))
+app.use(createPersistentLoggerMiddleware('zp.log'))
```
Comment (akolotov) [inline zp-relayer/index.ts:11]: May we make the path to the directory where the file is located configurable as so the corresponding directory can be mounted in the docker environment? In this case there will be no need to get the access to the file within the container.

File: zp-relayer/config.ts [hunk 2]
```diff
@@ -48,6 +48,7 @@ const config = {
     .map(s => parseInt(s, 10)),
   requireTraceId: process.env.RELAYER_REQUIRE_TRACE_ID === 'true',
   requireHTTPS: process.env.RELAYER_REQUIRE_HTTPS === 'true',
+  logIgnoreRoutes: (process.env.RELAYER_LOG_IGNORE_ROUTES || '').split(' ').filter(r => r.length > 0),
```
Comment (akolotov) [inline zp-relayer/config.ts:51]: Consider to make separators consistent for all configuration options. The space is the separator in the lists `RPC_URL` and `RELAYER_LOG_IGNORE_ROUTES`. Whereas the comma is the separator in `RELAYER_JSONRPC_ERROR_CODES`.

File: zp-relayer/services/loggerMiddleware.ts [hunk 3]
```diff
@@ -1,9 +1,31 @@
...
+  })
+}
+
+export function createConsoleLoggerMiddleware() {
+  return expressWinston.logger({
+    winstonInstance: logger,
+    level: 'debug',
+    ignoredRoutes: config.logIgnoreRoutes,
+    headerBlacklist: [
+      'accept',
```
Comment (akolotov) [inline zp-relayer/services/loggerMiddleware.ts:19]: Does it make sense to provide the ability to configure the list of the headers through a environment variable?

**Review by akolotov — CHANGES_REQUESTED** [review]

File: zp-relayer/services/loggerMiddleware.ts (hunk 3 above)
Comment (LyzeOfKiel) [reply zp-relayer/services/loggerMiddleware.ts:19]: Yeah, probably makes sense, leaving current header blacklist entries as default ones.

File: zp-relayer/config.ts (hunk 2 above)
Comment (LyzeOfKiel) [reply zp-relayer/config.ts:56]: Changed separator to space

File: zp-relayer/config.ts [hunk 4]
```diff
@@ -43,11 +47,16 @@ const config = {
...
-  relayerJsonRpcErrorCodes: (process.env.RELAYER_JSONRPC_ERROR_CODES || '-32603,-32002,-32005')
-    .split(',')
+  relayerJsonRpcErrorCodes: (process.env.RELAYER_JSONRPC_ERROR_CODES || '-32603 -32002 -32005')
+    .split(' ')
+    .filter(s => s.length > 0)
     .map(s => parseInt(s, 10)),
   requireTraceId: process.env.RELAYER_REQUIRE_TRACE_ID === 'true',
   requireHTTPS: process.env.RELAYER_REQUIRE_HTTPS === 'true',
+  logIgnoreRoutes: (process.env.RELAYER_LOG_IGNORE_ROUTES || '').split(' ').filter(r => r.length > 0),
+  logHeaderBlacklist: (process.env.RELAYER_LOG_HEADER_BLACKLIST || defaultHeaderBlacklist)
```
Comment (akolotov) [inline zp-relayer/config.ts:57]: please reflect the new environment variables in `CONFIGURATION.md`

**Review by akolotov — CHANGES_REQUESTED** [review]

File: zp-relayer/config.ts [hunk 5]
```diff
@@ -21,6 +24,7 @@ const config = {
   treeUpdateParamsPath: process.env.TREE_UPDATE_PARAMS_PATH || './params/tree_params.bin',
   transferParamsPath: process.env.TRANSFER_PARAMS_PATH || './params/transfer_params.bin',
   txVKPath: process.env.TX_VK_PATH || './params/transfer_verification_key.json',
+  requestLogPath: process.env.RELAYER_REQUEST_LOG_PATH || './zp.log',
```
Comment (akolotov) [inline zp-relayer/config.ts:27]: This also needs to be reflected in CONFIGURATION.md

**Review by akolotov — CHANGES_REQUESTED** [review]

**Review by akolotov — APPROVED** [review]

**Pull request merged**